x/wasm/handler.go

```diff
@@ -139,3 +144,20 @@ func handleMigration(ctx sdk.Context, k Keeper, msg *MsgMigrateContract) (*sdk.R
 	res.Events = append(events, ourEvent)
 	return res, nil
 }
+
+func handleUpdateContractAdmin(ctx sdk.Context, k Keeper, msg *MsgUpdateAdministrator) (*sdk.Result, error) {
+	if err := k.UpdateContractAdmin(ctx, msg.Contract, msg.Sender, msg.NewAdmin); err != nil {
+		return nil, err
+	}
+	events := filterMessageEvents(ctx.EventManager())
```

Comment:
This filtering stuff is only needed for the calls that dispatch to other handlers via the routers, as they would end up with ots of `sdk.EventTypeMessage` events that got merged together. I think you can just do

`events := ctx.EventManager().Events()`

here or something similar (like other modules do).